===================================================================
RCS file: /cvs/gcc/gcc/gcc/ada/a-stunau.adb,v
retrieving revision 1.5
retrieving revision 1.6
diff -u -r1.5 -r1.6
--- gcc/gcc/ada/a-stunau.adb	2003/10/21 13:41:54	1.5
+++ gcc/gcc/ada/a-stunau.adb	2005/01/18 22:00:12	1.6
@@ -48,16 +48,18 @@
 
             U_Ptr : constant Unbounded_String_Access := U'Unrestricted_Access;
             --  Unbounded_String is a controlled type which is always passed
-            --  by copy it is always safe to take the pointer to such object
-            --  here. This pointer is used to set the U.Reference value which
-            --  would not be possible otherwise as U is read-only.
+            --  by reference.  It is always safe to take the pointer to such
+            --  object here.  This pointer is used to set the U.Reference
+            --  value which would not be possible otherwise as U is read-only.
 
             Old : String_Access := U.Reference;
+            Ret : String_Access;
 
          begin
-            U_Ptr.Reference := new String'(U.Reference (1 .. U.Last));
+            Ret := new String'(U.Reference (1 .. U.Last));
+            U_Ptr.Reference := Ret;
             Free (Old);
-            return U.Reference;
+            return Ret;
          end;
       end if;
    end Get_String;
